java/dataset/src/main/java/org/apache/arrow/dataset/jni/NativeScanTask.java

```diff
@@ -35,7 +36,7 @@ public NativeScanTask(NativeScanner scanner) {
   }
 
   @Override
-  public BatchIterator execute() {
+  public ArrowReader execute() {
```

Comment:
Is there some numbers about to compare execution times for this change?

Reply:
I think it's kind of unfair to compare the performance between a `BatchIterator` and a `ArrowReader`. Because ArrowReader loads the buffers to a schema-aware VectorSchemaRoot. While the old code left that burden to caller. As a result it the code should be slower than old code (for example, in my local env, 642ms vs 379ms, 350MB parquet data). But the overall impact should be small in real project.